pkg/cosign/image_signer.go

```diff
@@ -1,34 +1,57 @@
 package cosign
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha2"
+	cosignremote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "k8s.io/client-go/kubernetes"
 
 	"github.com/buildpacks/lifecycle/platform"
 	"github.com/pkg/errors"
 	"github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
 )
 
-type SignFunc func(
-	ro *options.RootOptions, ko options.KeyOpts, signOpts options.SignOptions, imgs []string,
-) error
+type SignFunc func(*options.RootOptions, options.KeyOpts, options.SignOptions, []string) error
+
+type FetchSignatureFunc func(name.Reference, ...cosignremote.Option) (name.Tag, error)
+
+type BuilderSigner interface {
+	SignBuilder(context.Context, string, []corev1.Secret, authn.Keychain) ([]v1alpha2.SignaturePath, error)
+}
 
 type ImageSigner struct {
-	Logger   *log.Logger
-	signFunc SignFunc
+	Logger             *log.Logger
+	signFunc           SignFunc
+	fetchSignatureFunc FetchSignatureFunc
 }
```

Comment:
The `Logger` field doesn't seem to be used anywhere, can we get rid of it entirely?

Reply:
I think we can. It doesn't look like it was being used in the previously merged code either.

Reply:
Perhaps it was added for code parity with the `ImageSigner` in notary?
https://github.com/pivotal/kpack/blob/b323164d7c3f92c15b67b92ca0b1549383765369/pkg/notary/image_signer.go#L37-L41

Reply:
Removed the logger.